### PR TITLE
ENT-979. Make enterprise_id UUID for correct filtering

### PR DESCRIPTION
**Pull request by georgebabey**
As a string, only exact matches would properly filter. This was an issue when passing in a hyphenated UUID because the ids in the result store are just the 32 hex digits, no hyphens.

### CODE REVIEW

**Comment by georgebabey** [comment]
@UmanShahzad This is preventing the testing of ENT-979. See https://github.com/edx/edx-enterprise-data/pull/34/files#diff-ecf5d037d5373c24a5fb31678ddf6bffL141 for where we were converting to raw hex when looking up in Vertica. Rather than do that in the consumer, this change makes it more flexible on the API side to accept anything that can be a UUID.

File: enterprise_data/migrations/0005_auto_20180524_2204.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# Generated by Django 1.11.5 on 2018-05-24 22:04
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('enterprise_data', '0004_auto_20180508_1652'),
```
Comment (UmanShahzad) [inline enterprise_data/migrations/0005_auto_20180524_2204.py:11]: There is apparently some issue with finding this node in the tests:

```
django.db.migrations.exceptions.NodeNotFoundError: Migration enterprise_data.0005_auto_20180524_2204 dependencies reference nonexistent parent node ('enterprise_data', '0004_auto_20180508_1652')
```

Although I'm not sure why that is...

**Comment by UmanShahzad** [comment]
@georgebabey There may be some issue with travis caching or something. I'm going to try to re-open the PR and also try clearing the cache (if that's even possible).

**Comment by UmanShahzad** [comment]
@georgebabey I've totally cleared the TravisCI cache here in hopes that it'll fix the above issue.

**Comment by UmanShahzad** [comment]
@georgebabey I had to force re-push your branch after rebasing on master to get TravisCI to pick up some new changes. You may have to checkout this branch afresh locally.

**Review by UmanShahzad — APPROVED** [review]
:+1: Looks good.

**Pull request merged**